src/ipc/ipc4/helper.c

```diff
@@ -383,7 +384,21 @@ int ipc_comp_connect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 		return IPC4_INVALID_RESOURCE_ID;
 	}
 
-	buffer = list_first_item(&src->bsink_list, struct comp_buffer, source_list);
+	list_for_item(sink_list, &src->bsink_list) {
+		int src_queue = _connect[3];
```

Comment:
lets not do this, we should have a structure defined for this module queue and use it (rather than casting array members).

Reply:
sure, done